src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -283,7 +282,7 @@ protected override ImmutableArray<PendingBranch> RemoveReturns()
 
         protected virtual void ReportUnassignedOutParameter(ParameterSymbol parameter, SyntaxNode node, Location location)
         {
-            if (!_requireOutParamsAssigned && topLevelMethod == currentMethodOrLambda)
+            if (!_requireOutParamsAssigned && topLevelMethod == (object)currentMember)
```

Comment:
`ReferenceEquals` seems clearer here. I thought our `(object)` pattern was only for comparing to `null`.